src/components/intent-components/intent-list/intent-list.tsx

```diff
@@ -22,6 +22,42 @@ interface Props {
  */
 const IntentPanel: React.FC<Props> = ({ intents, onIntentClick }) => {
 
+  const [ searchValue, setSearchValue ] = React.useState("");
+
+  /**
+   * Event handler for search field change
+   * 
+   * @param event event
+   */
+  const onSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+    setSearchValue(value);
+  }
+
+  /**
+   * Renders a list of intents that match the search
+   */
+  const renderSearchedIntents = () => {
+
+    if (!intents) {
+      return null;
+    }
+
+    return (
+      <List>
+        {
+          intents.filter(item => item.name?.toLowerCase().includes(searchValue.toLowerCase())).map(intent => (
+            <InteractiveListItem
+              icon={ <IntentIcon/> }
+              title={ intent.name ?? "" }
+              onClick={ () => onIntentClick(intent) }
+            />
+          ))
+        }
+      </List>
+    );
+  }
```

Comment:
maybe better to use ternary expression
```suggestion
    { intents? <List>
        {
          intents.filter(item => item.name?.toLowerCase().includes(searchValue.toLowerCase())).map(intent => (
            <InteractiveListItem
              icon={ <IntentIcon/> }
              title={ intent.name ?? "" }
              onClick={ () => onIntentClick(intent) }
            />
          ))
        }
      </List> : null
    }
```